### DISCUSSION TITLE
Multichain dApp: An example of the power of soroban-react

### DISCUSSION THREAD

**Discussion by esteblock** [stellar/sorobounty-spectacular#17] [General]
Hello, In the context of my support of the @soroban-react library, after version 5.0.3 (Now in version 6.4.0) I worked on supporting multichain dapps.

![scaled50_2](https://github.com/stellar/sorobounty-spectacular/assets/2274485/9946716f-272b-4722-bae6-9ed874c4a0ac)
![scaled50_1](https://github.com/stellar/sorobounty-spectacular/assets/2274485/4fea9854-f865-43d9-9dc8-7eca53aca2c7)

This means that in the same frontend, the user can choose with which Network to work with (Very useful to mainnet / test).

I developed a **Multichain Dapp** in order to show the power of @soroban-react. I originally published my work [here](https://discord.com/channels/897514728459468821/1115726546162954411/1115726546162954411) and today I upgraded the dApp in order to  support Preview 10:

In the dapp, there is a contract that sets a **title**, and the user can change this by signing a transaction. However, there is one title for each supported network

Check the dApp and Tutorial here: https://github.com/esteblock/multichain-dapp

Main usage:

```javascript
import React from 'react'
import {SorobanReactProvider} from '@soroban-react/core';
import {futurenet, sandbox, standalone} from '@soroban-react/chains';
import {freighter} from '@soroban-react/freighter';
import {ChainMetadata, Connector} from "@soroban-react/types";
      
const chains: ChainMetadata[] = [sandbox, standalone, futurenet];
const connectors: Connector[] = [freighter()]
                          
                          
  export default function MySorobanReactProvider({children}:{children: React.ReactNode}) {
    return (
      <SorobanReactProvider
        chains={chains}
        appName={"Example Stellar App"}
        connectors={connectors}>
          {children}
      </SorobanReactProvider>
    )
  }
```

**Comment by briwylde08** [comment]
Thank you so much for your submission! We’ll review it and be in contact with you within the next week. 🎉

**Comment by briwylde08** [comment]
Hiya @esteblock! We've reviewed your submission! [🎉](https://emojipedia.org/party-popper/)

This is a _little_ out of scope for the Sorobounty Spectacular program but does showcase some neat functionality. Given this, your submission may qualify for an award of $300 worth of XLM. To receive that award, please fill out [this form](https://stellarform.typeform.com/to/HBjEyCJt) completely and accurately.

Thanks again! Please let us know if you have any questions.

Best,
Bri

**Comment by esteblock** [comment]
Hello @briwylde08 thanks! however, just to clarify... it is out of the scope?
It is a dApp, with
- a [smart contract code](https://github.com/esteblock/multichain-dapp/tree/main/contract/src)
- a [frontend and good explanation on how to run ](https://github.com/esteblock/multichain-dapp/tree/main)
- a [commented bash script to deploy](https://github.com/esteblock/multichain-dapp/blob/main/initialize.sh)

and the most important of all, showing how to build a multichain (Soroban mainnet/testnet) dapp!

thanks again

**Reply by briwylde08** [reply]
Heya! Okay, looking this over again, we can increase by $100 worth of XLM for a total of $400 worth of XLM. Here's a breakdown: 

- Dapp code ($300 worth of XLM)
- README & well-documented code ($100 worth of XLM) 

Please let me know if you have any other questions!

Best,
Bri

**Reply by esteblock** [reply]
Great, thanks!

**Comment by esteblock** [comment]
I filled the form :) (I also already filled it for Sorobanathon and First Light)

**Comment by briwylde08** [comment]
Got it! Thank you for providing the requested information! You’ve passed KYC and we’ve logged your award payment of $1,200 worth of XLM (combining your three submissions) which will be sent to the wallet you designated in the next week.

Thanks again and please let me know if you have any questions!

Best,
Bri
